packages/common/src/controllers/CheckoutController.ts

```diff
@@ -237,46 +258,37 @@ export default class CheckoutController {
 
     const { subscription } = useAccountStore.getState();
 
-    if (!subscription) return;
+    if (!subscription) return null;
 
     const response = await this.checkoutService.getSubscriptionSwitches({
       customerId: customerId,
       offerId: subscription.offerId,
     });
 
-    if (!response.responseData.available.length) return;
-
-    // create variable for `getOffer` to ensure it's typed in `Array#map` scope
-    const getOfferDelegate = this.checkoutService.getOffer;
-    const switchOffers = response.responseData.available.map((offer: SwitchOffer) => getOfferDelegate({ offerId: offer.toOfferId }));
-    const offers = await Promise.all(switchOffers);
+    if (!response.responseData.available.length) return null;
 
-    // Sort offers for proper ordering in "Choose Offer" modal when applicable
-    const offerSwitches = offers.sort((a, b) => a?.responseData.offerPrice - b?.responseData.offerPrice).map((item) => item.responseData);
-    useCheckoutStore.setState({ offerSwitches });
+    return response.responseData.available.map(({ toOfferId }) => toOfferId);
   };
 
-  switchSubscription = async (toOfferId: string, switchDirection: 'upgrade' | 'downgrade'): Promise<unknown> => {
+  switchSubscription = async () => {
+    const selectedOffer = useCheckoutStore.getState().selectedOffer;
+    const subscription = useAccountStore.getState().subscription;
     const { getAccountInfo } = useAccountStore.getState();
-
     const { customerId } = getAccountInfo();
 
+    if (!selectedOffer || !subscription) throw new Error('No offer selected');
     assertModuleMethod(this.checkoutService.switchSubscription, 'switchSubscription is not available in checkout service');
 
-    const { subscription } = useAccountStore.getState();
-    if (!subscription) return;
+    const switchDirection: 'upgrade' | 'downgrade' = determineSwitchDirection(subscription);
 
     const SwitchSubscriptionPayload = {
```

Comment:
Not touched in this PR, but perhaps we can fix this name as well 😄 
```suggestion
    const switchSubscriptionPayload = {
```

Reply:
Good idea :)